### PR TITLE
Update option step and create new option for select build system with…

### PR DESCRIPTION
**Pull request by manuelabarca**
…out arguments

### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version
2.0.0

### Context

I have created a new option within the steps so that the user can decide if he wants to use the legacy system or the new system, I have taken advantage of the same thing that the "options" option occupies to be able to send the arguments but in a more friendly way

Resolves: #50 

### Changes

Changes have been made to the step.yml and main.go to be able to use this new option that allows the user to switch between xcode build systems.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -259,6 +260,14 @@ func main() {
 		builder.SetCustomOptions(options...)
 	}
 
+	if configs.BuildSystem == "legacy" {
+		legacyQuery := "--buildFlag='-UseModernBuildSystem=0'"
+		builder.SetCustomOptions(legacyQuery)
+	} else if configs.BuildSystem == "modern" {
+		modernQuery := "--buildFlag='-UseModernBuildSystem=YES'"
```
Comment (ofalvai) [inline main.go:267]: For consistency, please use `YES` and `NO` or `0` and `1`, but let's not mix the two. According to this SO answer, both formats work: https://stackoverflow.com/a/51205222

File: step.yml [hunk 2]
```diff
@@ -119,19 +119,29 @@ inputs:
...
         Root directory of your Cordova project, where your Cordova config.xml exists.
       is_required: true
-  - options: --buildFlag="-UseModernBuildSystem=0"
+  - options:
     opts:
       title: "Options to append to the cordova-cli build command"
       description: |-
         Use this input to specify custom options, to append to the end of the cordova-cli build command.
 
         The new Xcode build system is now supported in cordova-ios@5.0.0 (https://github.com/apache/cordova-ios/issues/407).
```
Comment (ofalvai) [inline step.yml:128]: It's safe to delete this sentence too from the description.

File: step.yml [hunk 3]
```diff
@@ -119,19 +119,29 @@ inputs:
...
-        To use the legacy build system add `--buildFlag="-UseModernBuildSystem=0"` to the options string.
-
         Example:
         - `--browserify`
 
         `cordova build [OTHER_PARAMS] [options]`
+  - build_system: "modern"
+    opts:
+      title: "Build system"
```
Comment (ofalvai) [inline step.yml:135]:
```suggestion
      title: "Xcode build system"
```

File: step.yml [hunk 4]
```diff
@@ -119,19 +119,29 @@ inputs:
...
+      description: |-
+        The build system to use.
```
Comment (ofalvai) [inline step.yml:137]:
```suggestion
        The Xcode build system to use.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,7 +1,12 @@
-## Changelog (Current version: 1.2.1)
+## Changelog (Current version: 1.2.2)
 
 -----------------
 
+### 1.2.2 (2021 Oct 04)
```
Comment (ofalvai) [inline CHANGELOG.md:5]: We are changing the default behavior of the step here, so let's call this version 2.0.0. The legacy build system was the default, so it's a breaking change for those who used the default step inputs.

**Review by ofalvai — CHANGES_REQUESTED** [review]
Thank you for the contribution @manuelabarca! The changes look good, I just left a few minor comments about versioning and naming things. Looking forward to merge this PR!

File: step.yml (hunk 2 above)
Comment (manuelabarca) [reply step.yml:128]: Yes it's safe because, it's only default "text" in input in this option, with new option you've a new system in default.

**Comment by manuelabarca** [comment]
> Thank you for the contribution @manuelabarca! The changes look good, I just left a few minor comments about versioning and naming things. Looking forward to merge this PR!

The changes have been applied, thank you very much.

**Comment by manuelabarca** [comment]
PR it's ok ? Or wait other check ?

**Review by ofalvai — APPROVED** [review]

**Comment by ofalvai** [comment]
Yes, the PR looks good, I just had to fix a broken test in CI (it's not related to the changes in this PR).
Thank you for the contribution, we appreciate it! 🚀

**Pull request merged**